src/soc/ibm/power9/istep_8_10.c

```diff
@@ -0,0 +1,389 @@
+		uint8_t data[17] = {
+			0x42, 0x3E, 0x00, 0x60, 0x40, 0x40, 0x03, 0x03,
+			0x42, 0x03, 0x03, 0x40, 0x40, 0x60, 0x00, 0x3E,
+			0x42
+		};
+		and_or_scom(chip, addr, ~PPC_BITMASK(48, 54), PPC_PLACE(data[i], 48, 7));
+	}
+
+	/* *_TX_MODE1_PL */
+	for (i = 0; i < 17; i++) {
+		uint64_t addr = 0x8004040006010C3F + offset + 0x100000000*i;
+		and_scom(chip, addr, ~PPC_BIT(48));
+	}
+
+	/* *_TX_MODE2_PL */
+	for (i = 0; i < 17; i++) {
+		uint64_t addr = 0x80040C0006010C3F + offset + 0x100000000*i;
+		or_scom(chip, addr, PPC_BIT(62));
+	}
+
+	/* *_TX_BIT_MODE1_E_PL */
+	for (i = 0; i < 17; i++) {
+		uint64_t addr = 0x80043C0006010C3F + offset + 0x100000000*i;
+		uint16_t data[17] = {
+			0x000, 0x000, 0x01E, 0x01F, 0x00F, 0x07C, 0xC63, 0xE73,
+			0x000, 0xE73, 0xC63, 0x07C, 0x00F, 0x01F, 0x01E, 0x000,
+			0x000,
+		};
+		and_or_scom(chip, addr, ~PPC_BITMASK(48, 63), PPC_PLACE(data[i], 48, 16));
+	}
+
+	/* *_TX_BIT_MODE2_E_PL */
+	for (i = 0; i < 17; i++) {
+		uint64_t addr = 0x8004440006010C3F + offset + 0x100000000*i;
+		uint8_t data[17] = {
+			0x01, 0x7C, 0x7B, 0x0C, 0x5E, 0x10, 0x0C, 0x4E,
+			0x01, 0x4E, 0x0C, 0x10, 0x5E, 0x0C, 0x7B, 0x7C,
+			0x01,
+		};
+		and_or_scom(chip, addr, ~PPC_BITMASK(48, 54), PPC_PLACE(data[i], 48, 7));
+	}
+
+	// P9A_XBUS_0_RX[01]_RX_SPARE_MODE_PG
+	// 49 - IOF1_RX_RX0_RXCTL_CTL_REGS_RX_CTL_REGS_RX_PG_SPARE_MODE_1_ON
+	or_scom(chip, 0x8008000006010C3F + offset, PPC_BIT(49));
+
+	// P9A_XBUS_0_RX[01]_RX_ID1_PG
+	and_or_scom(chip, 0x8008080006010C3F + offset,
+		    ~PPC_BITMASK(48, 53),
+		    PPC_PLACE((group == 0 ? 0x00 : 0x01), 48, 6));
+
+	// P9A_XBUS_0_RX[01]_RX_CTL_MODE1_EO_PG
+	// 48 - IOF1_RX_RX0_RXCTL_CTL_REGS_RX_CTL_REGS_RX_CLKDIST_PDWN_OFF
+	and_scom(chip, 0x8008100006010C3F + offset, ~PPC_BIT(48));
+
+	// P9A_XBUS_0_RX[01]_RX_CTL_MODE5_EO_PG
+	// 51-53 - IOF1_RX_RX0_RXCTL_CTL_REGS_RX_CTL_REGS_RX_DYN_RECAL_INTERVAL_TIMEOUT_SEL_TAP5
+	// 54-55 - IOF1_RX_RX0_RXCTL_CTL_REGS_RX_CTL_REGS_RX_DYN_RECAL_STATUS_RPT_TIMEOUT_SEL_TAP1
+	and_or_scom(chip, 0x8008300006010C3F + offset,
+		    ~(PPC_BITMASK(51, 53) | PPC_BITMASK(54, 55)),
+		    PPC_PLACE(0x5, 51, 3) | PPC_PLACE(0x1, 54, 2));
+
+	// P9A_XBUS_0_RX[01]_RX_CTL_MODE7_EO_PG
+	and_or_scom(chip, 0x8008400006010C3F + offset,
+		    ~PPC_BITMASK(60, 63),
+		    PPC_PLACE(0xA, 60, 4));
+
+	// P9A_XBUS_0_RX0_RX_CTL_MODE23_EO_PG (same address for both groups)
+	// 55 - IOF1_RX_RX0_RXCTL_CTL_REGS_RX_CTL_REGS_RX_PEAK_TUNE_OFF
+	// 56 - IOF1_RX_RX0_RXCTL_CTL_REGS_RX_CTL_REGS_RX_LTE_EN_ON
+	// 59 - IOF1_RX_RX0_RXCTL_CTL_REGS_RX_CTL_REGS_RX_DFEHISPD_EN_ON
+	// 60 - IOF1_RX_RX0_RXCTL_CTL_REGS_RX_CTL_REGS_RX_DFE12_EN_ON
+	if (group == 0) {
+		and_or_scom(chip, 0x8008C00006010C3F,
+			~(PPC_BITMASK(48, 49) | PPC_BITMASK(55, 60)),
+			PPC_PLACE(0x1, 48, 2) | PPC_BIT(56) | PPC_PLACE(0x3, 57, 2) |
+			PPC_BIT(59) | PPC_BIT(60));
+	} else {
+		and_or_scom(chip, 0x8008C00006010C3F,
+			    ~PPC_BITMASK(48, 49), PPC_PLACE(0x1, 48, 2));
+	}
+
+	if (group == 1) {
```

Comment:
Can this be put inside `else` above? Or are there more than 2 groups?

Reply:
There are only 2 groups, but these registers aren't related. I've added a comment like for other registers instead.